### PR TITLE
Fixes #59: Test server uses an ephemeral port instead of binding to 8282

### LINKED ISSUE #59: Test server should use an ephemeral port to avoid "address already in use" errors
As a developer, one of the first things I like to do when I clone a project repo is run the test suite. Ideally I should be able to run `tox` at the command line and everything works.

The tests in sideboard/tests/test_server.py run a server that always binds to port 8282. If that port is already in use, the result is a bunch of "address already in use" errors.

Instead, the server should bind to port 0 to request an already open ephemeral port. The socket can then be inspected to determine what port it's listening on.

### PR DESCRIPTION
**Pull request by RobRuana**

### CODE REVIEW

File: sideboard/tests/__init__.py [hunk 1]
```diff
@@ -9,6 +11,26 @@
...
+    # the same port in the time between requesting an available port and
+    # actually using it. This is unlikely, and the worst that will happen is
+    # the tests fail on that particular test run.
+    #
+    # See https://eklitzke.org/binding-on-port-zero
+    #
+    # Ideally we could tell cherrypy to listen on port 0, and then inspect
+    # it to determine what port it's using, but cherrypy doesn't support that
+    # yet (other parts of cherrypy will try to use the port defined in
+    # 'cherrypy.server.socket_port' and end up failing on startup).
```
Comment (EliAndrewC) [inline sideboard/tests/__init__.py]: Great comment!  However, as a style note, anytime you have a giant comment right at the beginning of a function, it should probably be a triple-quoted docstring.

File: sideboard/tests/test_server.py [hunk 2]
```diff
@@ -19,10 +18,15 @@
 
 import sideboard.websockets
 from sideboard.lib import log, config, subscribes, notifies, services, cached_property, WebSocket
-from sideboard.tests import service_patcher, config_patcher
+from sideboard.tests import service_patcher, config_patcher, get_available_port
 from sideboard.tests.test_sa import Session
 
 
+available_port = get_available_port()
+config['cherrypy']['server.socket_port'] = available_port
```
Comment (kitsuta) [inline sideboard/tests/test_server.py]: Just curious, why are you updating config in two ways here?

**Comment by EliAndrewC** [comment]
+1 other than the one style comment about using a docstring rather than a series of repeated ``#`` one-line comments

File: sideboard/tests/test_server.py (hunk 2 above)
Comment (EliAndrewC) [inline sideboard/tests/test_server.py]: By the time this code is executed, we'll have already populated ``cherrypy.config`` with what's in our config file, so the configuration will already be living in two places, each of which we need to update.

Comment (kitsuta) [inline sideboard/tests/test_server.py]: I thought that might be the case. I guess we then still use our ConfigObj elsewhere?

Comment (RobRuana) [inline sideboard/tests/test_server.py]: I'm not sure of all the places it's being used, but I didn't want another developer in the future to try using it and wonder why it's not working. I wanted to make sure that `config['cherrypy']['server.socket_port']` always contains the value that we'd expect (principle of least astonishment).

**Pull request merged**